samples/cpp/visual_language_chat/visual_language_chat.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "load_image.hpp"
+#include <openvino/genai/vlm_pipeline.hpp>
+#include <openvino/runtime/intel_gpu/properties.hpp>
+
+bool print_subword(std::string&& subword) {
+    return !(std::cout << subword << std::flush);
+}
+
+int main(int argc, char* argv[]) try {
+    if (3 != argc) {
+        throw std::runtime_error(std::string{"Usage "} + argv[0] + " <MODEL_DIR> <IMAGE_FILE>");
+    }
+    ov::Tensor image = utils::load_image(argv[2]);
+    std::string device = "CPU";  // GPU can be used as well
+    ov::AnyMap enable_compile_cache;
+    if ("GPU" == device) {
+        // Cache compiled models on disk for GPU to save time on the
+        // next run. It's not beneficial for CPU.
+        enable_compile_cache.insert({ov::cache_dir("vlm_cache")});
+    }
```

Comment:
as we encourage users to modify samples code, do we need such if/else per-device conditions?

Reply:
There was someone requesting to describe how to activate cache. I think this place is a good opportunity.